File: src/main/java/api/routes/themes/ThemesService.java
package api.routes.themes;

import api.utils.ApiUtils;
import api.routes.themes.dto.ThemeDto;
import lombok.RequiredArgsConstructor;
import main.commands.slashcommands.commands.management.ThemeCommand;
import main.constants.RobertifyTheme;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Service;
import org.springframework.web.server.ResponseStatusException;

@Service
@RequiredArgsConstructor
public class ThemesService {

    public ResponseEntity<ThemeResponse> setTheme(ThemeDto themeDto) {
        final var server = ApiUtils.getGuild(themeDto.getServer_id());
        try {
            final var theme = RobertifyTheme.parse(themeDto.theme);
            new ThemeCommand().updateTheme(server, theme);
            return ResponseEntity.ok(ThemeResponse.builder().message("Successfully set the theme for " + server.getName() + " to: " + theme.name()).build());
        } catch (IllegalArgumentException e) {
            throw new ResponseStatusException(
                    HttpStatus.BAD_REQUEST,
                    "Invalid theme",
                    e
            );
        } catch (Exception e) {
            throw new ResponseStatusException(
                    HttpStatus.INTERNAL_SERVER_ERROR,
                    "Internal Server Error",
                    e
            );
        }
    }
}

File: src/main/kotlin/api/routes/themes/ThemesService.kt
package api.routes.themes

import api.models.response.GenericResponse
import api.models.response.OkResponse
import api.models.service.AbstractGuildService
import api.routes.themes.dto.ThemeDto
import main.commands.slashcommands.management.ThemeCommand
import main.constants.RobertifyTheme
import net.dv8tion.jda.api.sharding.ShardManager

class ThemesService(shardManager: ShardManager) : AbstractGuildService(shardManager) {

    suspend fun updateTheme(themeDto: ThemeDto): GenericResponse {
        val guild = guildUtils.getGuild(themeDto.server_id) ?: return noGuild(themeDto.server_id)

        ThemeCommand().updateTheme(guild, RobertifyTheme.parse(themeDto.theme), shardManager)
        return OkResponse(
            message = "Successfully set the theme for ${guild.name} to ${themeDto.theme.uppercase()}"
        )
    }

}